atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1124,18 +1138,33 @@ private int estimateSize(List<RowResult<Value>> rangeRows) {
         return estimatedSize;
     }
 
-    private <T, R extends ImmutableMap.Builder<Cell, T>> R getWithPostFiltering(
+    private <S extends Map<Cell, T>, T> S getWithPostFilteringSync(
             TableReference tableRef,
             Map<Cell, Value> rawResults,
-            R resultsAccumulator,
+            Function<Iterable<Map.Entry<Cell, T>>, S> mapFactory,
```

Comment:
on second thought/after seeing everything wired in, it's kinda weird to have this `mapFactory` (not really a map factory in its entirety either), let's just return the collection of entries that we get from `getWithPostFilteringAsync`, don't have to worry about weird generics, consumers can construct the map how they please.

Reply:
A much more elegant solution, fixed.